### PR TITLE
Run prow tests on kubernetes.

### PR DESCRIPTION
**Pull request by spxtr**
The dockerfile is similar to `pull-kubernetes-bazel`. Please do not merge for me, since I haven't pushed the image.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1062)

### CODE REVIEW

File: images/pull-test-infra-go-test/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    git \
+    wget \
+    python && \
+    apt-get clean
+
+RUN mkdir -p /workspace
+RUN wget https://storage.googleapis.com/golang/go1.7.3.linux-amd64.tar.gz && \
+    tar xf go1.7.3.linux-amd64.tar.gz && \
+    rm go1.7.3.linux-amd64.tar.gz
+ENV GOROOT /go
```
Comment (fejta) [inline images/pull-test-infra-go-test/Dockerfile]: Please collapse as many ENV commands as possible.

File: images/pull-test-infra-go-test/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+MAINTAINER spxtr@google.com
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    git \
+    wget \
+    python && \
+    apt-get clean
+
+RUN mkdir -p /workspace
```
Comment (fejta) [inline images/pull-test-infra-go-test/Dockerfile]: Please collapse as many RUN commands as possible.

File: prow/jobs.yaml [hunk 3]
```diff
@@ -176,3 +176,25 @@ kubernetes/test-infra:
...
   rerun_command: "@k8s-bot go test this"
   trigger: "@k8s-bot (go )?test this"
+  spec:
+    nodeSelector:
+      role: build
+    containers:
+    - name: builder
+      image: gcr.io/k8s-testimages/test-infra-go-test:0.0
+      command: ["/bin/bash", "-c"]
+      args: ["git clone https://github.com/kubernetes/test-infra && ./test-infra/jenkins/bootstrap.py --repo=k8s.io/test-infra --pull=${PULL_NUMBER} --job=pull-test-infra-go-test"]
```
Comment (fejta) [inline prow/jobs.yaml]: It seems like we are thinking of this image as a virtual machine? I would much prefer that we cleanup the interface to this image.

Comment (spxtr) [inline prow/jobs.yaml]: Might be interesting to use a kubernetes git volume to check out test-infra, set bootstrap.py as the image command, then just provide the arguments here.

**Comment by fejta** [comment]
Please leave a comment when ready for another review!

File: images/pull-test-infra-go-test/Dockerfile (hunk 1 above)
Comment (spxtr) [inline images/pull-test-infra-go-test/Dockerfile]: Done.

File: prow/jobs.yaml (hunk 3 above)
Comment (spxtr) [inline prow/jobs.yaml]: Okay, no command required.

**Comment by spxtr** [comment]
@fejta PTAL.

File: prow/jobs.yaml [hunk 4]
```diff
@@ -185,3 +185,23 @@ kubernetes/test-infra:
   context: Jenkins go test
   rerun_command: "@k8s-bot go test this"
   trigger: "@k8s-bot (go )?test this"
+  spec:
```
Comment (fejta) [inline prow/jobs.yaml]: FYI the verbosity in adding a job here will scale poorly to myriad jobs

Comment (spxtr) [inline prow/jobs.yaml]: Agreed, I'm going to make some changes to cut this down to pretty much nothing. For now this is what we've got.

**Pull request merged**